django/db/migrations/state.py

```diff
@@ -88,21 +90,52 @@ class ProjectState:
     FKs/etc. resolve properly.
     """
 
-    def __init__(self, models=None, real_apps=None):
+    def __init__(self, models=None, real_apps=None, related_fields=None):
         self.models = models or {}
         # Apps to include from main registry, usually unmigrated ones
         self.real_apps = real_apps or []
+        self.related_fields = related_fields or {}
```

Comment:
The map should get automatically populated from the provided models and not specified through initialization as that's an internal data structure.